src/gui/plugins/resource_spawner/ResourceSpawner.qml

```diff
@@ -301,11 +312,64 @@ Rectangle {
       Layout.minimumWidth: 300
       Layout.fillWidth: true
       spacing: 0
+      Rectangle {
+        id: searchSortBar
+        color: evenColor
+        height: 50
+        Layout.minimumWidth: 290
+        Layout.minimumHeight: 50
+        Layout.fillWidth: true
+        RowLayout {
+        id: rowLayout
+        spacing: 20
+        Rectangle {
+          color: "transparent"
+          height: 50
+          Layout.minimumWidth: 100
+          Layout.minimumHeight: 50
+          Layout.preferredWidth: (searchSortBar.width - 80) / 2
+          Layout.leftMargin: 10
+          TextField {
+            id: searchField
+            anchors.fill: parent
+            placeholderText: "Search"
+            leftPadding: 2
+            topPadding: 15
+            color: Material.theme == Material.Light ? "black" : "white"
+            onAccepted: {
```

Comment:
While doing some quick testing, it felt a bit awkward to have to press enter for the changes to take effect. That was especially confusing when I erased the search, didn't press enter, then selected another user, and no models showed. I think this could be at least `onEditingFinished` so it's triggered when losing focus.

I see that `TextField` doesn't have a signal that's triggered for every text change, but have you looked into `TextInput`? It looks like it has a [textEdited](https://doc.qt.io/qt-5/qml-qtquick-textinput.html#textEdited-signal) signal. Not sure how updating the results for every new letter would affect performance though.

Reply:
I changed the `TextField` to `TextInput` as the re-search for every key entered doesn't noticeably affect performance and search is still quick after every key entered.  The downside is that `TextInput` doesn't have the native underline for the input box nor does it have a `placeholderText` attribute.  I added the search icon next to the `TextInput` box and also made the box an offset color from the top bar that it's on so that it is more visible.  Additionally, I increased the minimum width of this plugin given the 2 pane format is more natural, I think, and the search and sort fits more naturally and doesn't feel squished.  Let me know what you think and if you think we can improve this anymore. 
![Screenshot from 2020-10-01 17-49-58](https://user-images.githubusercontent.com/8881852/94877223-bfde4400-040e-11eb-970d-a50e0c71724d.png)

Reply:
Works great!